tests/queries/test_contains.py

```diff
@@ -0,0 +1,49 @@
+from django.test import TestCase
+
+from .models import ChildObjectA, ObjectA, ObjectB, ProxyObjectA
+
+
+class ContainsTests(TestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.existing = ObjectA.objects.create(name='one')
+        cls.not_saved = ObjectA(name='two')
+        cls.wrong_model = ObjectB.objects.create(name='three', objecta=cls.existing, num=1)
+        cls.proxy_model_existing = ProxyObjectA.objects.create(name='four')
+
+    def test_non_model(self):
+        self.assertRaises(ValueError, ObjectA.objects.contains, object())
+
+    def test_values_querysets(self):
+        """ Make sure .values() and .values_list() QuerySets raises ValueError """
+        with self.assertRaises(ValueError):
```

Comment:
Use `assertRaisesMessage()`, e.g.
```suggestion
        msg = ...
        with self.assertRaisesMessage(TypeError, msg):
```